dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -14,6 +18,13 @@ class ScriptLevelsController < ApplicationController
   # This is set to twice the proxy max-age because of a bug in CloudFront.
   DEFAULT_PUBLIC_CLIENT_MAX_AGE = DEFAULT_PUBLIC_PROXY_MAX_AGE * 2
 
+  # Return true if request is one that can be publicly cached.
+  def self.is_cachable_request?(request)
+    script_id = request.params[:script_id]
+    script = Script.get_from_cache(script_id) if script_id
+    script && Gatekeeper.allows('public_caching_for_script', where: {script_name: script.name})
+  end
```

Comment:
Too bad they can't fix referer

> On Nov 19, 2015, at 18:49, Will Jordan <notifications@github.com> wrote:
> 
> In dashboard/app/controllers/script_levels_controller.rb:
> 
> > @@ -14,6 +18,13 @@ class ScriptLevelsController < ApplicationController
> >    # This is set to twice the proxy max-age because of a bug in CloudFront.
> >    DEFAULT_PUBLIC_CLIENT_MAX_AGE = DEFAULT_PUBLIC_PROXY_MAX_AGE * 2
> >  
> > +  # Return true if request is one that can be publicly cached.
> > +  def self.is_cachable_request?(request)
> > +    script_id = request.params[:script_id]
> > +    script = Script.get_from_cache(script_id) if script_id
> > +    script && Gatekeeper.allows('public_caching_for_script', where: {script_name: script.name})
> > +  end
> minor nit- consider standardizing on "cacheable" spelling for consistency with the HTTP/1.1+ RFCs?
> 
> (fun trivia around the spelling- HTTP/1.0 first used the spelling "cachable" in its text, and after some discussion HTTP/1.1 standardized on this spelling throughout most of its development, even using a "cachable" cache-control directive in early versions, until the 'e' was quietly find-replace inserted into HTTP/1.1 in a final draft.)
> 
> —
> Reply to this email directly or view it on GitHub.
>